<a href="https://colab.research.google.com/github/amallindask/data-science-2026/blob/main/Pertemuan2_Amallinda_250401020003.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

**IDENTITAS MAHASISWA**

---

Nama : Amallinda Sekar Kinasih

---

NIM : 250401020003

---

Kelas :IF403

In [31]:
import pandas as pd
import numpy as np

# 1 Muat dasaset Titanic
URL = "https://raw.githubusercontent.com/datasciencedojo/datasets/master/titanic.csv"
df = pd.read_csv(URL)
df

,PassengerId,Survived,Pclass,Name,Sex,Age,SibSp,Parch,Ticket,Fare,Cabin,Embarked
0,1,0,3,"Braund, Mr. Owen Harris",male,22.0,1,0,A/5 21171,7.2500,NaN,S
1,2,1,1,"Cumings, Mrs. John Bradley (Florence Briggs Th...",female,38.0,1,0,PC 17599,71.2833,C85,C
2,3,1,3,"Heikkinen, Miss. Laina",female,26.0,0,0,STON/O2. 3101282,7.9250,NaN,S
3,4,1,1,"Futrelle, Mrs. Jacques Heath (Lily May Peel)",female,35.0,1,0,113803,53.1000,C123,S
4,5,0,3,"Allen, Mr. William Henry",male,35.0,0,0,373450,8.0500,NaN,S
...,...,...,...,...,...,...,...,...,...,...,...,...
886,887,0,2,"Montvila, Rev. Juozas",male,27.0,0,0,211536,13.0000,NaN,S
887,888,1,1,"Graham, Miss. Margaret Edith",female,19.0,0,0,112053,30.0000,B42,S
888,889,0,3,"Johnston, Miss. Catherine Helen ""Carrie""",female,NaN,1,2,W./C. 6607,23.4500,NaN,S
889,890,1,1,"Behr, Mr. Karl Howell",male,26.0,0,0,111369,30.0000,C148,C


In [32]:
# 2 Eksplorasi Awal
print(df.shape, df.columns.tolist())
print(df.isnull().sum)

(891, 12) ['PassengerId', 'Survived', 'Pclass', 'Name', 'Sex', 'Age', 'SibSp', 'Parch', 'Ticket', 'Fare', 'Cabin', 'Embarked']
<bound method DataFrame.sum of      PassengerId  Survived  Pclass   Name    Sex    Age  SibSp  Parch  Ticket  \
0          False     False   False  False  False  False  False  False   False   
1          False     False   False  False  False  False  False  False   False   
2          False     False   False  False  False  False  False  False   False   
3          False     False   False  False  False  False  False  False   False   
4          False     False   False  False  False  False  False  False   False   
..           ...       ...     ...    ...    ...    ...    ...    ...     ...   
886        False     False   False  False  False  False  False  False   False   
887        False     False   False  False  False  False  False  False   False   
888        False     False   False  False  False   True  False  False   False   
889        False     False   Fal

In [33]:
# 3 filtering = Penumpang Wanita Kelas 1
wk1 = df[(df['Sex']=='female') & df['Pclass']==1]
print(wk1['Survived'].mean(), wk1['Age'].mean())

0.6848739495798319 27.59625668449198


In [34]:
# 4 Groupby — Tingkat Keselamatan per Kelas
hasil = df.groupby('Pclass').agg(
    total=('PassengerId','count'),
    selamat= ('Survived','mean'),
).round(3)
print(hasil)

        total  selamat
Pclass                
1         216    0.630
2         184    0.473
3         491    0.242


In [35]:
# PERTANYAAN KUIS
# Q1 : BERAPA JUMLAH BARIS DAN KOLOM DATASET TITANIC ?
print(f"Jumlah baris dan kolom: {df.shape}\n")
print(f"Kolom-kolom pada dataset: \n{df.columns.tolist()}\n")

Jumlah baris dan kolom: (891, 12)

Kolom-kolom pada dataset: 
['PassengerId', 'Survived', 'Pclass', 'Name', 'Sex', 'Age', 'SibSp', 'Parch', 'Ticket', 'Fare', 'Cabin', 'Embarked']



In [36]:
# Q2 : KOLOM MANA YANG MEMILIKI NILAI KOSONG (MISSING)?
print(f"Missing value pada dataset: \n {df.isnull().sum()}")
print(f"\nMissing value dalam persentase(%): \n {df.isnull().sum()}")

Missing value pada dataset: 
 PassengerId      0
Survived         0
Pclass           0
Name             0
Sex              0
Age            177
SibSp            0
Parch            0
Ticket           0
Fare             0
Cabin          687
Embarked         2
dtype: int64

Missing value dalam persentase(%): 
 PassengerId      0
Survived         0
Pclass           0
Name             0
Sex              0
Age            177
SibSp            0
Parch            0
Ticket           0
Fare             0
Cabin          687
Embarked         2
dtype: int64


In [37]:
# Q3 : BERAPA PERSEN PENUMPANG YANG SELAMAT(SURVIVED=1)?
# selamat = wk1[wk1['Survived']==1]
print(f"Persentase seluruh penumpang selamat : {df['Survived'].mean()*100}%")
print(f"Persentase penumpang wanita yang selamat di Kelas 1 : {wk1['Survived'].mean()*100} %")


Persentase seluruh penumpang selamat : 38.38383838383838%
Persentase penumpang wanita yang selamat di Kelas 1 : 68.4873949579832 %


In [38]:
# Q4 : FILTERING PENUMPANG WANITA DARI KELAS 1 (Pclass=1). BERAPA JUMLAHNYA DAN BERAPA RATA RATA USIANYA?
# wk1 = df[(df['Sex']=='female') & df['Pclass']==1]
print(f"{ wk1 }")
print(f"Jadi, jumlah penumpang wanita di Kelas 1 : {wk1.shape[0]} orang")
print(f"Rata-rata usia penumpang kelas 1 adalah : {wk1['Age'].mean()} tahun")


     PassengerId  Survived  Pclass  \
1              2         1       1   
2              3         1       3   
3              4         1       1   
8              9         1       3   
10            11         1       3   
..           ...       ...     ...   
879          880         1       1   
882          883         0       3   
885          886         0       3   
887          888         1       1   
888          889         0       3   

                                                  Name     Sex   Age  SibSp  \
1    Cumings, Mrs. John Bradley (Florence Briggs Th...  female  38.0      1   
2                               Heikkinen, Miss. Laina  female  26.0      0   
3         Futrelle, Mrs. Jacques Heath (Lily May Peel)  female  35.0      1   
8    Johnson, Mrs. Oscar W (Elisabeth Vilhelmina Berg)  female  27.0      0   
10                     Sandstrom, Miss. Marguerite Rut  female   4.0      1   
..                                                 ...     ...   ... 

In [39]:
# Q5 : GUNAKAN GROUPBY UNTUK MENGHITUNG TINGKAT KESELAMATAN (MEAN SURVUVED) PER KELAS PENUMPANG. KELAS MANA YANG PALING TINGGI?
print(f"Tingkat Keselamatan Per kelas :")
print(hasil)

Tingkat Keselamatan Per kelas :
        total  selamat
Pclass                
1         216    0.630
2         184    0.473
3         491    0.242


# Q6 : BUAT ANALISIS TAMBAHAN BERDASARKAN VARIABEL LAIN YANG ANDA ANGGAP MENARIK. SERTA INTERPRETASI SINGKAT.
Saya melakukan analisis terhadap variabel Age untuk melihat hubungannya dengan Survived. Dari hasil pengamatan, penumpang yang berusia lebih muda, terutama anak-anak, memiliki peluang selamat lebih besar dibandingkan penumpang yang berusia lebih tua. Hal ini kemungkinan karena pada saat evakuasi, anak-anak dan perempuan lebih diprioritaskan untuk diselamatkan terlebih dahulu. Jadi, usia dapat menjadi salah satu faktor yang memengaruhi tingkat keselamatan penumpang Titanic.

**Kesimpulan**

---
Pada pertemuan ini mempelajari cara melakukan eksplorasi data menggunakan Pandas, seperti melihat ukuran dataset, mengidentifikasi missing value, melakukan filtering data, serta menggunakan fungsi groupby untuk menganalisis hubungan antara variabel dengan tingkat keselamatan penumpang.

---

Temuan utama menunjukkan bahwa dataset Titanic memiliki 891 baris dan 12 kolom. Beberapa kolom memiliki nilai kosong, terutama Cabin (687 data), Age (177 data), dan Embarked (2 data). Hasil analisis juga menunjukkan bahwa kelas penumpang memengaruhi peluang keselamatan, di mana kelas 1 memiliki tingkat keselamatan tertinggi, diikuti kelas 2, dan kelas 3. Selain itu, penumpang wanita kelas 1 memiliki tingkat keselamatan yang relatif tinggi, yaitu sekitar 68,49%.

---

Yang dapat menjadi pertanyaan adalah Apakah penumpang dengan tiket lebih mahal benar-benar memiliki akses lebih baik ke sekoci penyelamat?

---

Karena pada data ini menunjukkan bahwa kelas sosial, jenis kelamin, dan usia merupakan faktor-faktor signifikan yang memengaruhi tingkat keselamatan penumpang Titanic, mencerminkan ketidaksetaraan yang terjadi pada saat tragedi tersebut.